fastify.js

```diff
@@ -603,6 +604,16 @@ function build (options) {
     // we might want to do some hard debugging
     // here, let's print out as much info as
     // we can
+    req.id = genReqId(req)
+    req.log = res.log = log.child({ reqId: req.id })
+    req.originalUrl = req.url
+
+    req.log.info({ req }, 'incoming request')
+
+    res._startTime = now()
+    res.on('finish', onResFinished)
+    res.on('error', onResFinished)
```

Comment:
> Nope, the customized 404 routes uses the same code of the standard routes, so they are covered. Even the 404 fallback (which is a very particular case) is covered.

For that to be true, shouldn't this code be inside the `defaultRoute` function?

Regular handler: `router.lookup -> fastify`
Regular notFound handler: `router.lookup -> defaultRoute -> fourOhFour.lookup -> custom notFoundHandler / basic404`
Fallback notFound handler: `router.lookup -> defaultRoute -> fourOhFour.lookup -> fourOhFourFallback`

Reply:
Nope, because a successful `fourOhFour.lookup` will "land" in the `fastify` function.

Reply:
Oh of course! You're right @delvedor :)